File: platforms/android/runtime/core/src/main/java/io/nativeblocks/runtime/api/provider/block/defaults/RootBlock.kt
package io.nativeblocks.runtime.api.provider.block.defaults

import androidx.compose.foundation.layout.Column
import androidx.compose.runtime.Composable
import io.nativeblocks.runtime.api.provider.block.BlockContext
import io.nativeblocks.runtime.api.provider.block.NONE_INDEX

@Composable
internal fun RootBlock(blockContext: BlockContext) {
    val slots = blockContext.block.slots

    val contentSlot = slots["content"]

    Column {
        if (contentSlot != null) {
            blockContext.onSubBlock.invoke(
                blockContext.block.subBlocks.orEmpty(),
                contentSlot,
                NONE_INDEX,
                null,
            )
        }
    }
}

File: platforms/ios/runtime/Sources/NativeblocksRuntime/api/provider/block/defaults/RootBlock.swift
import Foundation
import SwiftUI

internal struct RootBlock: View {
    var blockContext: BlockContext
    @Environment(\.nativeWindowWidthClass) var windowManager

    var body: some View {
        let slots = blockContext.block.slots
        let contentSlot = slots["content"]

        VStack {
            if let contentSlot {
                blockContext.onSubBlock(blockContext.block.subBlocks ?? [:], contentSlot, NONE_INDEX, nil)
            }
        }
    }
}
